src/backend/joanie/core/models/products.py

```diff
@@ -484,6 +484,14 @@ class Order(BaseModel):
         null=True,
         encoder=OrderPaymentScheduleEncoder,
     )
+    credit_card = models.ForeignKey(
+        to="payment.CreditCard",
+        verbose_name=_("credit card"),
+        related_name="orders",
+        on_delete=models.SET_NULL,
+        blank=True,
+        null=True,
+    )
```

Comment:
Should we add a db constraint to ensure that a fee order has a credit card ?

Reply:
Yup that what I meaned. Sorry.

> Should we add a db constraint to ensure that a not free order has a credit card ?

Reply:
From my understanding, we can't do something like that:
```python
            models.CheckConstraint(
                check=(~models.Q(product__price=0) | models.Q(credit_card__isnull=False)),
                name="payment_method_required_for_paid_order",
                violation_error_message="Payment method is required for paid orders",
            ),
```
We get this error:
```
django.core.exceptions.FieldError: Joined field references are not permitted in this query
```
We can't go through `product` to get its price.

Reply:
We can add a check in order.save() or order.clean()
WDYT @jbpenrath @lunika @jonathanreveille  ?

Reply:
Indeed, while working with db constraints I've realized that it was only possible to make checkups on the model attributes but not its relations.

I would add the check in `order.clean()`.

Reply:
The entire lifecycle of a credit card should be refacto IMO. We can't just add this field, there is more to do and IMO it's probably out of the scope of this refacto.